b.php

```diff
@@ -61,34 +61,43 @@ function b_init() {
   $command = array(
     'options' => array(
       'root' => 'Backdrop root folder',
+      'url' => 'Backdrop site URL (as defined in sites.php)',
       'drush' => 'Use .drush.inc files instead. Drupal 7 drush commands compatibility.',
       'y' => 'Force Yes to all Yes/No questions',
       'yes' => 'Force Yes to all Yes/No questions',
-      'd' => 'Debug mode',
+      'd' => 'Debug mode on',
       'debug' => 'Debug mode on',
     ),
   );
   b_get_command_args_options($arguments, $options, $command);
 
-  $_backdrop_root = FALSE;
+  b_init_globals();
+
+  // Get root directory.
   if (isset($options['root'])) {
-    if (file_exists($options['root'] . '/settings.php')) {
-      $_backdrop_root = $options['root'];
-    }
+    $_backdrop_root = b_find_root($options['root'], FALSE);
   }
   else {
-    $path = getcwd();
-    if (file_exists($path . '/settings.php')) {
-      $_backdrop_root = $path;
-    }
+    $_backdrop_root = b_find_root(getcwd());
   }
 
-  b_init_globals();
-
   if ($_backdrop_root) {
+    // Get site directory.
+    if (isset($options['url'])) {
+      $_backdrop_site = b_find_site_by_url($options['url'], $_backdrop_root);
+    }
+    else {
+      $_backdrop_site = b_find_site_by_path(getcwd(), $_backdrop_root);
+    }
+
     chdir($_backdrop_root);
     $full_path = getcwd();
     define('BACKDROP_ROOT', $full_path);
+    $_SERVER['HTTP_HOST'] = basename($full_path);
+    if ($_backdrop_site) {
+      define('BACKDROP_SITE', $_backdrop_site);
+      $_SERVER['HTTP_HOST'] = $_backdrop_site;
+    }
     require_once 'core/includes/bootstrap.inc';
```

Comment:
Wouldn't this fail, if one calls b from a multisite directory (which includes the sites settings.php).

I think, we should make this (all of) b's _require_once_ more robust regarding the base dir for including core files.

Reply:
Thanks for the review, I'll take a look tonight.

Reply:
@indigoxela Can you tell me exactly what command you're running and from which directory?

If I run `b st` from `backdrop/sites/my-site`, I get:

`Backdrop CMS Installation detected` (i.e. it found `settings.php`)
`Backdrop root: backdrop`
`Backdrop site: my-site`